src/wallet/mod.rs

```diff
@@ -253,32 +261,91 @@ where
             .map_err(|_| Error::ScriptDoesntHaveAddressForm)
     }
 
-    // Return the the last previously derived address for `keychain` if it has not been used in a
-    // received transaction. Otherwise return a new address using [`Wallet::get_new_address`].
-    fn get_unused_address(&self, keychain: KeychainKind) -> Result<AddressInfo, Error> {
+    // Return vector of unused address indexes
+    fn get_unused_key_indexes(&self, keychain: KeychainKind) -> Result<Vec<u32>, Error> {
```

Comment:
Wouldn't it be simpler and more efficient just to do `has_address_been_used` and return a `bool`? You are iterating over all the the scripts and checking them all even when you may only be interested in a particular one. The alternative would be to return a `HashSet` of used addresses or something similar.